bin/filter_sce_rds.R

```diff
@@ -62,13 +75,38 @@ if(!dplyr::between(opt$prob_compromised_cutoff, 0, 1)){
 # read in unfiltered rds file
 unfiltered_sce <- readr::read_rds(opt$unfiltered_file)
 
+# check if ADT data is present, and prep if so
+if (is.null(opt$adt_barcode_file)) {
+  ambient_profile <- NULL
+} else { 
+
+  # Create data frame of ADTs and their target types
+  # If `target_type` column is not present, assume all ADTs are targets
+  adt_barcode_df <- readr::read_tsv(
+    opt$adt_barcode_file, 
+    # if only 2 columns exist, only the first two col_names will be used
+    col_names = c("name", "barcode", "target_type")
+  ) 
+  if (!"target_type" %in% names(adt_barcode_df)) {
+    adt_barcode_df$target_type <- "target"
+  } 
+
+  # assign and check name for this alternative experiment
+  adt_exp <- opt$adt_name
+  if (!adt_exp %in% altExpNames(unfiltered_sce)) {
+    stop("Given named ADT alternative experiment not present in unfiltered SCE.")
+  }
```

Comment:
I'm a big proponent of failing fast in code (not in orgs) so I would move the check up to before reading any files. 
```suggestion
  # assign and check name for this alternative experiment
  adt_exp <- opt$adt_name
  if (!adt_exp %in% altExpNames(unfiltered_sce)) {
    stop("Given named ADT alternative experiment not present in unfiltered SCE.")
  }
  
  # Create data frame of ADTs and their target types
  # If `target_type` column is not present, assume all ADTs are targets
  adt_barcode_df <- readr::read_tsv(
    opt$adt_barcode_file, 
    # if only 2 columns exist, only the first two col_names will be used
    col_names = c("name", "barcode", "target_type")
  ) 
  if (!"target_type" %in% names(adt_barcode_df)) {
    adt_barcode_df$target_type <- "target"
  } 

```

Reply:
Secondary thought: we don't actually use the `adt_barcode_df` until much later... do we want to save reading until that point? 

Alternatively, we could use it here to check that the names match the altExp rownames. That seems a check worth doing to make sure we have the right pairing.

Related: we should definitely add the `target_type` column to the altExp rowData. (I don't know if the barcode seq is already there, but it also doesn't seem important to add)